clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -378,11 +378,12 @@ bool SemaSYCL::isDeclAllowedInSYCLDeviceCode(const Decl *D) {
   if (const FunctionDecl *FD = dyn_cast<FunctionDecl>(D)) {
     const IdentifierInfo *II = FD->getIdentifier();
 
-    // Allow __builtin_assume_aligned and __builtin_printf to be called from
-    // within device code.
-    if (FD->getBuiltinID() &&
-        (FD->getBuiltinID() == Builtin::BI__builtin_assume_aligned ||
-         FD->getBuiltinID() == Builtin::BI__builtin_printf))
+    // Allow __builtin_assume_aligned, printf and __builtin_printf to be called
+    // from within device code.
+    const auto BuiltinID = FD->getBuiltinID();
+    if (BuiltinID && (BuiltinID == Builtin::BI__builtin_assume_aligned ||
+                      BuiltinID == Builtin::BIprintf ||
```

Comment:
I would like to tag @AaronBallman and @gmlueck here for their feedback.

If I remember correctly, we specifically decided *not* to accept regular `printf` here, because we don't support it and didn't really want to jump through the hoops to make it work.

Even though we now have a SPIR-V extension to represent `printf` with non-constant format string, I don't think that all of our backends support it already (otherwise it would have been enabled by default).

Moreover, `printf` in SPIR-V corresponds to OpenCL C `printf`, which has a different format string specification, meaning that users of regular `printf` may encounter some bugs.

`experimental::printf` was introduced on purpose so that we differentiate it from a standard `printf` so that we can document all the quirks and pre-requisites our `printf` has without adding broken support for standard C/C++ `printf`.

Reply:
I've noticed [`sycl-varargs-cconv.cpp`](https://github.com/jchlanda/llvm/blob/jakub/cuda_printf/clang/test/SemaSYCL/sycl-varargs-cconv.cpp#L66) failing in CI and was thinking that perhaps `printf` was deliberately not supported (even though, for some cases at least, it just works). Perhaps I've been too hasty with this patch, am happy to close it.

Reply:
> I would like to tag @AaronBallman and @gmlueck here for their feedback.
> 
> If I remember correctly, we specifically decided _not_ to accept regular `printf` here, because we don't support it and didn't really want to jump through the hoops to make it work.
> 
> Even though we now have a SPIR-V extension to represent `printf` with non-constant format string, I don't think that all of our backends support it already (otherwise it would have been enabled by default).
> 
> Moreover, `printf` in SPIR-V corresponds to OpenCL C `printf`, which has a different format string specification, meaning that users of regular `printf` may encounter some bugs.
> 
> `experimental::printf` was introduced on purpose so that we differentiate it from a standard `printf` so that we can document all the quirks and pre-requisites our `printf` has without adding broken support for standard C/C++ `printf`.

You are remembering what I am remembering. :-)

Reply:
Yes, this is my recollection also.

Reply:
OK, I've discuss this offline with @Naghasan and it seems like we are converging on the fact that we want to keep it disabled. Closing this PR.